go/vt/topo/k8stopo/file.go

```diff
@@ -243,11 +242,16 @@ func (s *Server) List(ctx context.Context, filePathPrefix string) ([]topo.KVInfo
 	if len(nodes) == 0 {
 		return results, topo.NewError(topo.NoNode, filePathPrefix)
 	}
+	rootPrefix := filepath.Join(s.root, filePathPrefix)
 	for _, node := range nodes {
-		if strings.HasPrefix(node.Data.Value, filePathPrefix) {
+		if strings.HasPrefix(node.Data.Key, rootPrefix) {
+			out, err := unpackValue([]byte(node.Data.Value))
+			if err != nil {
+				return results, convertError(err, node.Data.Key)
+			}
 			results = append(results, topo.KVInfo{
 				Key:     []byte(node.Data.Key),
-				Value:   []byte(node.Data.Value),
+				Value:   out,
 				Version: KubernetesVersion(node.GetResourceVersion()),
```

Comment:
@mattlord FYI since it looks like your name is on the `git blame`, but there were 3 separate bugs here in the K8s topo `List` implementation. It wasn't checking against the correct path prefix (the `root` was ignored), it was checking against the `.Value` instead of the `.Key` and the `.Value` was not decompressed to match that compression is transparent.

This PR added some basic tests for `List` that uncovered all those issues. It does make me wonder if anyone is really using the K8s topo if there's bugs like this in there?

Reply:
Good catch, thank you! I'm also not sure if anyone is using it.

Reply:
not sure either. It was added by @carsonoid who may be able to tell us. or @derekperkins